cli/sql.go

```diff
@@ -100,11 +100,23 @@ func runInteractive(db *sql.DB, dbURL string) {
 			break
 		}
 
+		tl := strings.TrimSpace(l)
+		if len(stmt) == 0 && len(tl) == 0 {
+			// Empty line at beginning, simply continue.
+			// However, we don't simply continue after the first line since
+			// we may be in the middle of a string literal where empty lines
+			// may be significant.
```

Comment:
psql runs pg's SQL lexer while reading input at the prompt. For us that would mean invoking stuff from sql/parser in cli/sql . perhaps this would help splitting statements too. I'd suggest filling a separate issue as possible enhancement.

Nathan VanBenschoten <notifications@github.com> schreef op 7 februari 2016 23:32:46 CET:
>> @@ -100,11 +100,23 @@ func runInteractive(db *sql.DB, dbURL string) {
>>  			break
>>  		}
>>  
>> +		tl := strings.TrimSpace(l)
>> +		if len(stmt) == 0 && len(tl) == 0 {
>> +			// Empty line at beginning, simply continue.
>> +			// However, we don't simply continue after the first line since
>> +			// we may be in the middle of a string literal where empty lines
>> +			// may be significant.
>
>Interestingly, Postgres doesn't add blanks lines to the history even if
>it's in the middle of statement as long as it's not part of a string
>literal. Might be worth adding a TODO to make the prompt a little more
>intelligent about when it is and is not in the middle of a string
>literal. Here's an example from `psql`. Note the modified prompt
>(`nathan-#` vs `nathan'# `) while in strings.
>
>```
>nathan=# SELECT 
>nathan-# 
>nathan-# 
>nathan-# '
>nathan'# 
>nathan'# 
>nathan'# '
>nathan-# 
>nathan-# ;
> ?column? 
>----------
>         +
>         +
>         +
> 
>(1 row)
>
>nathan=# SELECT 
>'
>
>
>'
>;
>```
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/cockroachdb/cockroach/pull/4210/files#r52126277

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.